hudi-utilities/src/main/java/org/apache/hudi/utilities/streamer/StreamSync.java

```diff
@@ -333,29 +333,35 @@ public StreamSync(HoodieStreamer.Config cfg, SparkSession sparkSession,
   }
 
   /**
-   * Refresh Timeline.
+   * Creates a meta client for the table. If the table does not yet exist, it will be initialized.
    *
+   * @param refreshTimeline when set to true, loads the active timeline and updates the {@link #commitsTimelineOpt} and {@link #allCommitsTimelineOpt} values
+   * @return a meta client for the table
    * @throws IOException in case of any IOException
    */
-  public void refreshTimeline() throws IOException {
+  public HoodieTableMetaClient initializeMetaClient(boolean refreshTimeline) throws IOException {
     if (storage.exists(new StoragePath(cfg.targetBasePath))) {
       try {
         HoodieTableMetaClient meta = HoodieTableMetaClient.builder()
             .setConf(HadoopFSUtils.getStorageConfWithCopy(conf))
             .setBasePath(cfg.targetBasePath)
             .setPayloadClassName(cfg.payloadClassName)
             .setRecordMergerStrategy(null)
+            .setTimeGeneratorConfig(HoodieTimeGeneratorConfig.newBuilder().fromProperties(props).withPath(cfg.targetBasePath).build())
```

Comment:
Nice catch~

Reply:
@danny0405 I couldn't find a way to pass this in to the properties based builder used here though https://github.com/apache/hudi/pull/11853/files#diff-8367f95ad8e071faaa1fa8931acb5189582fffb2ab9d5cd5e3566a7d9f622736R407

Reply:
@danny0405 should we revisit all meta client builders to make sure the time generator config is properly set if required?

Reply:
Yeah, we should, the in-memory time generator is sufficient for most of the use cases, we only need to check the invoker who used the meta client to generate instant timestamps.